flink-clients/src/main/java/org/apache/flink/client/program/ContextEnvironment.java

```diff
@@ -74,15 +72,6 @@ public JobExecutionResult execute(String jobName) throws Exception {
 		return lastJobExecutionResult;
 	}
 
-	@Override
-	public String getExecutionPlan() throws Exception {
-		Plan plan = createProgramPlan("unnamed job");
-
-		OptimizedPlan op = ClusterClient.getOptimizedPlan(client.compiler, plan, getParallelism());
```

Comment:
I think that now we can make the `client.compiler` field `private` to avoid leaking information about internal state.

Reply:
And maybe other `ClusterClient` fields can become `private` as well.